lib/cli.js

```diff
@@ -63,38 +67,62 @@ module.exports = function broccoliCLI(args) {
 
       guardOutputDir(outputDir);
 
-      const builder = getBuilder({ brocfilePath: options.brocfilePath });
+      const builder = getBuilder(options);
       const outputTree = new TreeSync(builder.outputPath, outputDir);
+      const watcher = new Watcher(builder);
 
-      builder
-        .build()
-        .then(() => outputTree.sync())
-        .finally(() => builder.cleanup())
-        .then(() => process.exit(0))
+      watcher.on('buildSuccess', () => {
+        outputTree.sync();
+        messages.onBuildSuccess(builder);
+
+        if (!options.watch) {
+          watcher.quit();
+        }
+      });
+      watcher.on('buildFailure', messages.onBuildFailure);
+
+      function cleanupAndExit() {
+        return watcher.quit();
+      }
+
+      process.on('SIGINT', cleanupAndExit);
+      process.on('SIGTERM', cleanupAndExit);
+
+      actionPromise = watcher
+        .start()
+        .catch(err => console.log((err && err.stack) || err))
+        .finally(() => {
+          builder.cleanup();
+          process.exit(0);
+        })
         .catch(err => {
-          // Should show file and line/col if present
-          if (err.file) {
-            console.error('File: ' + err.file);
-          }
-          console.error(err.stack);
-          console.error('\nBuild failed');
+          console.log('Cleanup error:');
+          console.log((err && err.stack) || err);
           process.exit(1);
         });
     });
 
   program.parse(args || process.argv);
 
-  if (!actionPerformed) {
+  if (!actionPromise) {
     program.outputHelp();
     process.exit(1);
   }
+
+  return actionPromise || RSVP.resolve();
 };
 
 function getBuilder(options) {
   if (!options) {
     options = {};
   }
-  return new broccoli.Builder(broccoli.loadBrocfile(options.brocfilePath));
+  const brocfile = broccoli.loadBrocfile(options.brocfilePath);
+  process.chdir(options.cwd || brocfile.baseDir);
```

Comment:
I would have thought we would `chdir` before loading the brocfile, that way `cwd` is consistent before we touch user code.

Reply:
@stefanpenner was thinking about that and have mixed feelings. 

consider using `--brocfile-path` with `--cwd` together, example:
`broccoli serve --cwd <project-path> --brocfile-path ./Brocfile.development.js`
should the brocfile path be resolved to `<project-path>/Brocfile.development.js` or to `<current working dir>/Brocfile.development.js` ? 


another issue is `findup` in `load_brocfile.js` - will it be understandable that findup will be affected by `--cwd`?

Reply:
hmm, I'll have to think a bit more..

Reply:
I think `--cwd` should not affect the CLI arguments, but should affect things like `process.cwd()` in userland code. My mental model is that CLI parameters are evaluated in the context of the shell. This ensures the shells autocomplete works regardless of the `--cwd`. As if the user wants to change `cwd` in the shell, they should just do that...

So:

```sh
broccoli --cwd somewhere/else --brocfile-path ./Brocfile.development.js
```

```js
// Brocfile.js
process.cwd() // should be --cwd
```

Reply:
:+1: I do like the argument with shell autocompletion